sql/core/src/test/java/test/org/apache/spark/sql/execution/sort/RecordBinaryComparatorSuite.java

```diff
@@ -273,7 +273,7 @@ public void testBinaryComparatorWhenSubtractionIsDivisibleByMaxIntValue() throws
     insertRow(row1);
     insertRow(row2);
 
-    assert(compare(0, 1) < 0);
+    assert(compare(0, 1) > 0);
```

Comment:
So, do you mean this is wrong before this PR?

Reply:
The change definitely changes the ordering, as bytes are compared in a different order.

Reply:
> So, do you mean this is wrong before this PR?

`RecordBinaryComparator` is used as a local sort comparator in `RoundRobinPartition` to make sure the order of each records are the same after rerun. But the *relative order* is not important.